readthedocs/projects/tasks/builds.py

```diff
@@ -479,7 +479,7 @@ def on_failure(self, exc, task_id, args, kwargs, einfo):
         # Known errors in our application code (e.g. we couldn't connect to
         # Docker API). Report a generic message to the user.
         if isinstance(exc, BuildAppError):
-            message_id = BuildAppError.GENERIC_WITH_BUILD_ID
+            message_id = exc.message_id
```

Comment:
I would use a fallback here in case we are raising an exception without `message_id` for some reason.

```suggestion
            message_id = exc.message_id or BuildAppError.GENERIC_WITH_BUILD_ID
```

Reply:
message_id is a required field, if it's none somewhere in the code, that's a bug we should fix not let it pass silently.